cypress-io/cypress
# fix: Adding overload to screenshot to fix type errors when overwriting this command.

### Additional details
- Why was this change necessary? 

When trying to overwrite this command copying what is mentioned in the docs will throw a type error. The example provided in the docs is correct however the type definitions need to have an overload case that accounts for when screenshot is chained from another cypress command. Currently there is no overload that matches this case.

- What is affected by this change?
type definitions for the `cy.screenshot` method.

### Steps to test
when copying this[ code snippet](https://docs.cypress.io/api/cypress-api/custom-commands#Overwrite-screenshot-command) from the cypress docs

```
Cypress.Commands.overwrite(
  'screenshot',
  (originalFn, subject, fileName, options) => {
    // call another command, no need to return as it is managed
    cy.get('.app')
      .should('be.visible')

      // overwrite the default timeout, because screenshot does that internally
      // otherwise the `then` is limited to the default command timeout
      .then({ timeout: Cypress.config('responseTimeout') }, () => {
        // return the original function so that cypress waits for it
        return originalFn(subject, fileName, options)
      })
  }
)
```
We should no longer see the type errors outlined in this post from the docs.

https://github.com/cypress-io/cypress-documentation/pull/5323#issuecomment-1603468367

### How has the user experience changed?
Should no longer see type errors when trying to overwrite the screenshot command. See the linked comment above for screenshots of the error messages.

### PR Tasks


- [ ] Have tests been added/updated? I dont think this is needed but let me know if i'm wrong on that.
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? https://github.com/cypress-io/cypress-documentation/pull/5323#event-9619113541
I made a pr to update the docs for this, however the docs were mostly correct already, the type definitions are what's the issue here.
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)? No api changes, just type changes.
